src/features/wishlist/wishlistStore.ts

```diff
@@ -0,0 +1,48 @@
+import {create} from 'zustand';
+import {persist} from 'zustand/middleware';
+
+export interface WishlistItem {
+    productId: number;
+}
+
+export interface WishlistState {
+    wishlist: WishlistItem[];
+    addToWishlist: (id: number) => void;
+    removeFromWishlist: (id: number) => void;
+    clearWishlist: () => void;
+}
+
+export const useWishlistStore = create<WishlistState>()(
+    persist(
+        (set) => ({
+            wishlist: [],
+            addToWishlist: (id: number) => set((state) => {
+                const existingWishlistItem = state.wishlist.find(item => item.productId === id);
+
+                let newWishlist;
```

Comment:
Same advice I gave for newCart, strongly type your variable
i.e `let newWishlist : WishlistItem[] = []`